src/renderer/components/data-settings/data-settings.js

```diff
@@ -882,87 +886,116 @@ export default defineComponent({
 
       const requiredKeys = [
         'playlistName',
-        'videos'
+        'videos',
       ]
 
       const optionalKeys = [
+        'description',
+        'createdAt',
+      ]
+
+      const ignoredKeys = [
         '_id',
+        'title',
+        'type',
         'protected',
-        'removeOnWatched'
+        'lastUpdatedAt',
+        'lastPlayedAt',
+        'removeOnWatched',
+
+        'thumbnail',
+        'channelName',
+        'channelId',
+        'playlistId',
+        'videoCount',
       ]
 
       const requiredVideoKeys = [
         'videoId',
         'title',
         'author',
         'authorId',
-        'published',
         'lengthSeconds',
         'timeAdded',
-        'isLive',
-        'type',
+
+        // `playlistItemId` should be optional for backward compatibility
+        // 'playlistItemId',
       ]
 
-      playlists.forEach(async (playlistData) => {
+      playlists.forEach((playlistData) => {
         // We would technically already be done by the time the data is parsed,
         // however we want to limit the possibility of malicious data being sent
         // to the app, so we'll only grab the data we need here.
 
         const playlistObject = {}
 
         Object.keys(playlistData).forEach((key) => {
-          if (!requiredKeys.includes(key) && !optionalKeys.includes(key)) {
+          if ([requiredKeys, optionalKeys, ignoredKeys].every((ks) => !ks.includes(key))) {
             const message = `${this.$t('Settings.Data Settings.Unknown data key')}: ${key}`
             showToast(message)
           } else if (key === 'videos') {
             const videoArray = []
             playlistData.videos.forEach((video) => {
-              let hasAllKeys = true
-              requiredVideoKeys.forEach((videoKey) => {
-                if (!Object.keys(video).includes(videoKey)) {
-                  hasAllKeys = false
-                }
-              })
+              const videoPropertyKeys = Object.keys(video)
+              const videoObjectHasAllRequiredKeys = requiredVideoKeys.every((k) => videoPropertyKeys.includes(k))
 
-              if (hasAllKeys) {
+              if (videoObjectHasAllRequiredKeys) {
                 videoArray.push(video)
               }
             })
 
             playlistObject[key] = videoArray
+          } else if (ignoredKeys.includes(key)) {
+            // Do nothing for keys to be ignored
           } else {
```

Comment:
```suggestion
          } else if (!ignoredKeys.includes(key)) {
            // Do nothing for keys to be ignored
```

<sub>**Obligatory disclaimer:** I am prone to lexical errors when writing in the GitHub UI</sub>

Reply:
https://github.com/FreeTubeApp/FreeTube/pull/4234/commits/36e7ed883afc535d0004df0445a46da2e780ab94